### ISSUE TITLE
Upgrade CypherDSL to 2021.2.4 to fix query generation.

### ISSUE THREAD

**Issue by venkat125** [spring-projects/spring-data-neo4j#2386]
Similar to the following test model:
[similar model from Integration Test](https://github.com/spring-projects/spring-data-neo4j/blob/3bbff33353b4e79a709bb8ff08bf96f3d5900a18/src/test/java/org/springframework/data/neo4j/integration/issues/gh2323/GH2323IT.java)

Apart from the `knows` relationship to have another relationship for different label, added 
```
@Relationship(type = "MOTHER_TONGUE_IS", direction = Relationship.Direction.OUTGOING)
    @Property("motherTongue")
    private KnowsMtEntity motherTongue;

```
```
@RelationshipProperties
@EqualsAndHashCode(callSuper = true)
@Data
public class KnowsMtEntity extends RBaseEntity {

    @Id
    @GeneratedValue private Long id;

    @TargetNode private Language language;

}
```

for `(p:PERSON)-[:MOTHER_TONGUE_IS]->(l:LANGUAGE))`
when i used `Repository.findById(id)`
I get the `motherTongue ` relationship is being merged to `knows `relationship array variable in person object on response. But its intermittent. When i made the SDN logs enabled i see the query differs on every execution.
Case 1: Log with wrong response:

> MATCH (n:`MALE`:`EXTRA_LABEL_1`:`EXTRA_LABEL_2`) WHERE n.nid = $__id__ RETURN n{.created_on, .d_o_b, .is_active, .mid, .name, .nid, .profession, .updated_on, __nodeLabels__: labels(n), __internalNeo4jId__: id(n), MALE_MOTHER_TONGUE_IS_LANGUAGE: [(n)-[MALE__relationship__LANGUAGE:`**MOTHER_TONGUE_IS**`]->(n_motherTongue:`LANGUAGE`) | n_motherTongue{.name, __nodeLabels__: labels(n_motherTongue), __internalNeo4jId__: id(n_motherTongue), MALE__relationship__LANGUAGE}], MALE_KNOWS_LANGUAGE: [(n)-[MALE__relationship__LANGUAGE]->(n_knows:`LANGUAGE`) | n_knows{.name, __nodeLabels__: labels(n_knows), __internalNeo4jId__: id(n_knows), MALE__relationship__LANGUAGE}]}
> 2021-09-24 17:40:47.330 DEBUG 13504 --- [p-nio-80-exec-2] o.s.d.n.c.t.Neo4jTransactionManager      : Initiating transaction commit
> response :{"mid":"21M0002","name":"abcd","profession":"IT","dateOfBirth":"xxxxx","updated":"xxxxxxx","is_active":true,"motherTongue":{"id":67,"language":{"name":"english"}},"knows":[{"id":67,"language":{"name":"english"}},{"id":65,"speak":true,"read":true,"write":true,"language":{"name":"german"}}],"nid":"0507cb3f-f790-11eb-91d7-c90c6f7f0f17|21M0002"}

Case 2: Log with correct response:

> MATCH (n:`MALE`:`EXTRA_LABEL_1`:`EXTRA_LABEL_2`) WHERE n.nid = $__id__ RETURN n{.created_on, .d_o_b, .is_active, .mid, .name, .nid, .profession, .updated_on, __nodeLabels__: labels(n), __internalNeo4jId__: id(n), MALE_KNOWS_LANGUAGE: [(n)-[MALE__relationship__LANGUAGE:`**KNOWS**`]->(n_knows:`LANGUAGE`) | n_knows{.name, __nodeLabels__: labels(n_knows), __internalNeo4jId__: id(n_knows), MALE__relationship__LANGUAGE}], MALE_MOTHER_TONGUE_IS_LANGUAGE: [(n)-[MALE__relationship__LANGUAGE]->(n_motherTongue:`LANGUAGE`) | n_motherTongue{.name, __nodeLabels__: labels(n_motherTongue), __internalNeo4jId__: id(n_motherTongue), MALE__relationship__LANGUAGE}]}
> 2021-09-24 17:41:46.010 DEBUG 8276 --- [p-nio-80-exec-2] o.s.d.n.c.t.Neo4jTransactionManager      : Initiating transaction commit
> response :{"mid":"21M0002","name":"abcd","profession":"IT","dateOfBirth":"xxxxx","updated":"xxxx","is_active":true,"motherTongue":{"id":67,"language":{"name":"english"}},"knows":[{"id":65,"speak":true,"read":true,"write":true,"language":{"name":"german"}}],"nid":"0507cb3f-f790-11eb-91d7-c90c6f7f0f17|21M0002"}

Even though the label for relationship in the query is not added properly for both the query (case 1:KNOWS) label missing and in (case 2: MOTHER_TONGUE_IS) label missing.
However, the case 2 response is correct.
Note: no code change between two cases. Just stopped and started the application.

**Comment by venkat125** [comment]
sorry for late reply.
I've checked and seems now its taking both the relationship labels correctly in debug query with desired result using the SNAPSHOT version.

> MATCH (n:`MALE`:`EXTRA_LABEL_1`:`EXTRA_LABEL_2`) WHERE n.nid = $__id__ RETURN n{.created_on, .d_o_b, .is_active, .mid, .name, .nid, .profession, .updated_on, __nodeLabels__: labels(n), __internalNeo4jId__: id(n), MALE_KNOWS_LANGUAGE: [(n)-[MALE__relationship__LANGUAGE:`**KNOWS**`]->(n_knows:`LANGUAGE`) | n_knows{.name, __nodeLabels__: labels(n_knows), __internalNeo4jId__: id(n_knows), MALE__relationship__LANGUAGE}], MALE_MOTHER_TONGUE_IS_LANGUAGE: [(n)-[MALE__relationship__LANGUAGE:`**MOTHER_TONGUE_IS**`]->(n_motherTongue:`LANGUAGE`) | n_motherTongue{.name, __nodeLabels__: labels(n_motherTongue), __internalNeo4jId__: id(n_motherTongue), MALE__relationship__LANGUAGE}]}
> 2021-10-01 16:46:13.495 DEBUG 11508 --- [p-nio-80-exec-2] o.s.d.n.c.t.Neo4jTransactionManager      : Initiating transaction commit
> response:{"mid":"21M0002","name":"abcd","profession":"IT","dateOfBirth":"xxxx","updated":"xxxx","is_active":true,"motherTongue":{"id":67,"language":{"name":"english"}},"knows":[{"id":65,"speak":true,"read":true,"write":true,"language":{"name":"german"}}],"nid":"0507cb3f-f790-11eb-91d7-c90c6f7f0f17|21M0002"}

Thanks @meistermeier and Team
If u don't mind, could u comment the PR of `CypherDSL to 2021.2.4` for this fix please?

**Comment by meistermeier** [comment]
Thanks for the positive response.
It would be this commit https://github.com/neo4j-contrib/cypher-dsl/commit/cc8f018a675c9818504feeb8156ff977ba649e13 that ports a change from the 2021.3.x line back to 2021.2.x (released as 2021.2.4).
You could also enforce the usage of 2021.2.4 if you define the dependency to the CypherDSL and set the version in maven or gradle explicitly.
```
<dependency>
  <groupId>org.neo4j</groupId>
  <artifactId>neo4j-cypher-dsl</artifactId>
  <version>2021.2.4</version>
</dependency>
```
```
implementation 'org.neo4j:neo4j-cypher-dsl:2021.2.4'
```

**Comment by meistermeier** [comment]
Fixed with 8aaee8063c7b0c6b06443b372114ceef0d0f0679 (for 6.1.x)
Main branch is already fixed because it uses CypherDSL 2021.3.x that has the fix already applied.
